agent/config/config.go

```diff
@@ -311,6 +216,157 @@ func get(args []string, cfg *Config, l *logrus.Entry) (configFileF string, err e
 	return //nolint:nakedret
 }
 
+func setDefaults(cfg *Config, l *logrus.Entry) {
+	if cfg == nil {
+		return
+	}
+
+	// set default values
+	if cfg.WindowConnectedTime == 0 {
+		cfg.WindowConnectedTime = time.Hour
+	}
+
+	configurePaths(cfg, l)
+	configureNetworkDefaults(cfg)
+
+	if cfg.Server.Address != "" {
+		if _, _, e := net.SplitHostPort(cfg.Server.Address); e != nil {
+			host := cfg.Server.Address
+			cfg.Server.Address = net.JoinHostPort(host, "443")
+			l.Infof("Updating PMM Server address from %q to %q.", host, cfg.Server.Address)
+		}
+	}
+
+	// enabled cross-component PMM_DEBUG and PMM_TRACE take priority
+	if b, _ := strconv.ParseBool(os.Getenv("PMM_DEBUG")); b {
+		cfg.Debug = true
+	}
+	if b, _ := strconv.ParseBool(os.Getenv("PMM_TRACE")); b {
+		cfg.Trace = true
+	}
+}
+
+var (
+	defaultListenSocketJSONFilename = "pmm-agent.sock"
+	defaultListenSocketGRPCFilename = "pmm-agent-grpc.sock"
+)
+
+func configureNetworkDefaults(cfg *Config) {
+	configureListenDefaults(cfg)
+
+	if cfg.Ports.Min == 0 {
+		cfg.Ports.Min = 42000 // for minimal compatibility with PMM Client 1.x firewall rules and documentation
+	}
+	if cfg.Ports.Max == 0 {
+		cfg.Ports.Max = 51999
+	}
+}
+
+func configureListenDefaults(cfg *Config) {
+	if cfg.ListenSocket == "" {
+		switch {
+		case cfg.ListenAddress == "" && cfg.ListenPort == 0:
+			cfg.ListenSocket = filepath.Join(cfg.Paths.PathsBase, defaultListenSocketJSONFilename)
+			if cfg.ListenSocketGRPC == "" {
+				cfg.ListenSocketGRPC = filepath.Join(cfg.Paths.PathsBase, defaultListenSocketGRPCFilename)
+			}
```

Comment:
this looks like the same code in `configureListenGRPCDefaults()`, is it a duplicate? or it has to be here too for some reason?

Reply:
we need this part in both functions.
The agent listens both for grpc and json. grpc is always listening on a socket while json could be either socket or ip:port.
If json is ip:port, we need to configure the grpc defaults in `configureListenGRPCDefaults()` because `configureListenDefaults()` does not take care of the default in this case.